server/src/main/resources/org/elasticsearch/index/IndexVersions.csv

```diff
@@ -131,3 +131,4 @@
 8.15.1,8512000
 8.15.2,8512000
 8.15.3,8512000
+8.15.4,8512000
```

Comment:
Yes. Otherwise the test would fail with a message similar to the following
```
java.lang.AssertionError: 
Expected: (<[8.14.4]> or <[8512000]> or <[8.14.0-8.14.4]>)
     but: was <[8.14.0-8.14.3]>
```
See #111777 for an example. 

@mark-vieira Could you please take a look at this change and make sure it is OK to update it manually like this?

Reply:
I think I understand why this change needs to be made, I just don't understand why it's happening in this PR. Shouldn't this be added by the release process?

Reply:
> Shouldn't this be added by the release process?

I have the same question. That's reason I requested @mark-vieira 's review. Maybe I should have tagged @elastic/es-delivery 
Also ping @thecoop since #111777 is assigned to you. Thank you!

Reply:
The build automation I think is not quite working properly with the multiple release branches we have - there's also https://github.com/elastic/elasticsearch/issues/114972. I need to look at this properly, but these should be added in a separate PR so we can confirm that it is actually the right thing to do at the right time (and backport it appropriately too)

Reply:
Thanks @thecoop !
I also noticed the missing versions are now added in main 85b2bab2e38409449a37adc9408902cbf79f8c8f
So these changes are indeed obsolete. I have removed them with e14dcb022c8ca159518beb7f80585b7f5bf5cc5a

Reply:
Errr the tests now [complain](https://gradle-enterprise.elastic.co/s/7rwqwan4xgxki/tests/task/:qa:full-cluster-restart:v8.16.1%23bwcTest/details/org.elasticsearch.upgrades.FullClusterRestartIT/testSnapshotRestore%20%7Bcluster=OLD%7D?top-execution=1) about 8.16.1 which is not yet in the versions file. 😿 

```
Expected: (<[8.16.1]> or <[8518000]> or <[8.16.0]>) |  
    but: was <[8.16.0-8.16.1]>
```

Cannot really move this PR forward without the versions to be fixed first.